tests/rptest/scale_tests/rpk_many_partitions_test.py

```diff
@@ -0,0 +1,144 @@
+        super(RpkManyPartitionsTest,
+              self).__init__(test_ctx,
+                             num_brokers=8,
+                             extra_rp_conf={
+                                 'topic_partitions_per_shard': 10000,
+                                 'topic_memory_per_partition': None,
+                             },
+                             *args,
+                             **kwargs)
+        self._rpk = RpkTool(self.redpanda)
+
+    @cluster(num_nodes=12)
+    def test_rpk_simple(self):
+        """
+        This tests rpk against a cluster with a significant number of partitions.
+        This is a simple test, and it does not take into account the stress of
+        a large number of consumers.
+        """
+        msg_count = 1000
+        partition_count_1 = 50000
+        partition_count_2 = 80000
+        group_name = "g1"
+        topic_spec = TopicSpec(name="foo",
+                               partition_count=partition_count_1,
+                               replication_factor=3)
+
+        self._rpk.create_topic(topic=topic_spec.name,
+                               partitions=topic_spec.partition_count,
+                               replicas=topic_spec.replication_factor)
+
+        # Produce 1k messages of 16KB size to the topic
+        # using `rpk topic produce` and then consuming
+        # from it.
+        def _produce_and_consume():
+            rpk_producer = RpkProducer(self.test_context,
+                                       self.redpanda,
+                                       topic=topic_spec.name,
+                                       msg_size=16384,
+                                       msg_count=msg_count,
+                                       printable=True,
+                                       produce_timeout=60)
+            rpk_producer.start()
+
+            # Consume messages from the topic with
+            # `rpk topic consume`.
+            rpk_consumer = RpkConsumer(self.test_context,
+                                       self.redpanda,
+                                       save_msgs=False,
+                                       retry_sec=0.5,
+                                       topic=topic_spec.name,
+                                       group=group_name)
+            rpk_consumer.start()
+
+            wait_until(lambda: rpk_consumer.message_count >= msg_count,
+                       timeout_sec=120,
+                       backoff_sec=5,
+                       err_msg="RPK failed to fetch all messages")
+
+            self.logger.debug(
+                f"RPK successfully fetched {rpk_consumer.messages}")
+
+            # `rpk topic consume` is in a loop that will
+            # run forever unless we call stop here
+            rpk_consumer.stop()
+
+            # `rpk topic produce` should be done already but
+            # we call wait and stop just to be sure
+            rpk_producer.wait()
+            rpk_producer.stop()
```

Comment:
Not really, I stop these at the end of the `_produce_and_consume` since this method is used twice in the test and the producer/consumer is created inside the method.

Maybe I can return the producer/consumer in the method and stop them at the end of the test, wdyt?

Reply:
> Maybe I can return the producer/consumer in the method and stop them at the end of the test, wdyt?

This sounds good. I notice that you call `_produce_and_consume` twice. If you return the producer and consumer, then you can stop and restart them in the test.